coralproject/talk
# Reintroducing E2E testing with Puppeteer

## What does this PR do?
- Reintroduces e2e with Pupeteer
- Adds missing classes and prop-types

## Summary and Concerns 
I’ve been using Puppeteer for a couple of days now and I ran into a couple of issues to set our e2e test environment. One of them ([Clicking element inside iFrame does not give any response · Issue #684 · GoogleChrome/puppeteer · GitHub](https://github.com/GoogleChrome/puppeteer/issues/684)) has been a blocker to develop tests for our `pym` handlers.  I’ve been able to come with a solution/workaround for it but I realised that there are a lot of features that interact with our `pym` service. 

Since Puppeteer does not provide a framework around testing we have to build the suite around it. The good thing about this is that we have more control over it. 

Another thing is that Puppeteer works only with Chrome, so we lose the API for testing cross-browser support that Selenium provides.

We can either explore more solutions for this, exploring other frameworks, other options or wait until they solve this bug.

If we move forward with this I plan on using more test architecture stuff like, page objects and better handling for selectors and errors. (Like screenshots for errors, etc.)

## How do I test this PR?
- run `yarn e2e`
- You should see: 
<img width="750" alt="screen shot 2017-10-02 at 12 04 06 am" src="https://user-images.githubusercontent.com/1401559/31063095-7274e22c-a706-11e7-9220-f44d1cc51333.png">


https://github.com/coralproject/talk/pull/188
